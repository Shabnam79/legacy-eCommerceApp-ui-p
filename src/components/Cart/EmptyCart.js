import React, { useEffect } from 'react'
export default function EmptyCart() {

    useEffect(() => {
        window.scrollTo(0, 0);
    }, []);
    return (
        <div className="container mt-5" data-testid="empty-cart">
            <div className="row">
                <div className="col-10 mx-auto text-center text-title">
                    <h1>Your cart is currently empty</h1>
                </div>
            </div>
        </div>
    );
}
