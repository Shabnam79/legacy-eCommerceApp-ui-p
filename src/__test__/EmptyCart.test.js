import React from 'react';
import { render } from '@testing-library/react';
import EmptyCart from '../../src/components/Cart/EmptyCart';

describe('EmptyCart component', () => {
    it('renders without crashing', () => {
        const { getByTestId } = render(<EmptyCart />);
        const emptyCartElement = getByTestId('empty-cart');
        expect(emptyCartElement).toBeInTheDocument();
    });

    it('displays the correct message', () => {
        const { getByText } = render(<EmptyCart />);
        const emptyCartMessage = getByText('Your cart is currently empty');
        expect(emptyCartMessage).toBeInTheDocument();
    });

    it('has the correct CSS class', () => {
        const { getByTestId } = render(<EmptyCart />);
        const emptyCartElement = getByTestId('empty-cart');
        expect(emptyCartElement).toHaveClass('container mt-5');
    });

    //ensures that the rendered component matches the previous snapshot.
    //This is useful for detecting unexpected changes in the component's appearance over time
    it('matches snapshot', () => {
        const { asFragment } = render(<EmptyCart />);
        expect(asFragment()).toMatchSnapshot();
    });
});
