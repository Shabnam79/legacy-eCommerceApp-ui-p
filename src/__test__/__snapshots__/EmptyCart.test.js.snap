// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyCart component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-5"
    data-testid="empty-cart"
  >
    <div
      class="row"
    >
      <div
        class="col-10 mx-auto text-center text-title"
      >
        <h1>
          Your cart is currently empty
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
